bulk-user-setup/bulk-user-update-CCD-1239.sh

```diff
@@ -0,0 +1,497 @@
+  json=$(convert_input_file_to_json "${filepath_input_original}")
+  check_exit_code_for_error $? "$json"
+
+  # input file read ok ...
+  # ... so move it to backup location
+  mv "$filepath_input_original" "$filepath_input_newpath" 2> /dev/null
+  if [ $? -eq 0 ]; then
+    echo "Moved input file to backup location: ${BOLD}${filepath_input_newpath}${NORMAL}"
+  else
+    echo "${RED}ERROR: Aborted as unable to move input file to backup location:${NORMAL} ${filepath_input_original}"
+    exit 1
+  fi
+
+  # write headers to output file
+  echo "operation,email,firstName,lastName,roles,inviteStatus,idamResponse,idamUserJson,timestamp" >> "$filepath_output_newpath"
+
+  # strip JSON into individual items then process in a while loop
+  echo $json | jq -r -c '.[]' \
+      |  \
+  ( success_counter=0;skipped_counter=0;fail_counter=0;total_counter=0;
+    while IFS= read -r user; do
+      total_counter=$((total_counter+1))
+
+      # extract CSV fields from json to use in output
+      local email=$(echo $user | jq --raw-output '.idamUser.email')
+      local inviteStatus=$(echo $user | jq --raw-output '.extraCsvData.inviteStatus')
+      local operation=$(echo $user | jq --raw-output '.extraCsvData.operation')
+      local rolesToAdd=$(echo $user | jq --raw-output '.idamUser.rolesToAdd')
+      local rolesToRemove=$(echo $user | jq --raw-output '.idamUser.rolesToRemove')
+
+      if [ "$inviteStatus" != "SUCCESS" ]; then
+        
+        # load formatted user JSON ready to send to IDAM 
+        idamUserJson=$(echo $user | jq -c --raw-output '.idamUser')
+        if [ "$operation" == "add" ]; then
+          # make call to IDAM to add user
+          submit_response=$(submit_user_registation "$idamUserJson")
+        elif [ "$operation" == "update" ]; then
+          # get user id and roles from IDAM
+          local rawReturnedValue=$(get_user "$email")
+          submit_response=$rawReturnedValue
+
+          # Prevent further API calls if User ID cannot be returned
+          if [[ $rawReturnedValue != *"HTTP-"* ]]; then 
+            local userId=$(echo $rawReturnedValue | jq --raw-output '.id')
+
+            # make call to IDAM to update roles for existing user
+            submit_response=$(add_user_role "$userId" "$caseworker_civil_id")
+            if [[ $submit_response != *"HTTP-"* ]]; then 
+              submit_response=$(add_user_role "$userId" "$caseworker_civil_solicitor_id")
+            fi
+          else
+            submit_response=$(echo "$rawReturnedValue")
+          fi
+        fi
+        # seperate submit_user_registation reponse
+        IFS=$'\n'
+        local response_array=($submit_response)
+        local inviteStatus=${response_array[0]}
+        local idamResponse=${response_array[1]}
+        if [ $inviteStatus == "SUCCESS" ]; then
+          # SUCCESS:
+          success_counter=$((success_counter+1))
+          echo "${total_counter}: ${email}: ${GREEN}${inviteStatus}${NORMAL}"
```

Comment:
Removed idamResponse from successful output as it returns the user object rather than a http status code